File: docs/examples/java/output/parts/UserManagementService_findUserOrThrow.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 203-209
// symbol: UserManagementService#findUserOrThrow
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls users.get
    private User findUserOrThrow(String userId) {
        User user = users.get(userId);
        if (user == null) {
            throw new UserNotFoundException("ユーザーID '" + userId + "' が見つかりません");
        }
        return user;
    }

File: docs/examples/python/output/parts/UserManagementService__find_user_or_throw.py
# code2map fragment (non-buildable)
# original: docs/examples/python/user_management_service.py
# lines: 227-231
# symbol: UserManagementService#_find_user_or_throw
# notes: references dataclasses.dataclass, re, typing.Optional; calls UserNotFoundException, self._users.get
    def _find_user_or_throw(self, user_id: str) -> User:
        user = self._users.get(user_id)
        if user is None:
            raise UserNotFoundException(f"ユーザーID '{user_id}' が見つかりません")
        return user
